ts/modules/dumbbell.src.ts

```diff
@@ -0,0 +1,567 @@
+ * @sample {highcharts} highcharts/series-dumbbell/styled-mode-dumbbell/
+ *         Styled mode
+ *
+ * @extends      plotOptions.arearange
+ * @product      highcharts highstock
+ * @excluding    fillColor, fillOpacity, lineWidth, stack, stacking,
+ *               stickyTracking, trackByArea
+ * @since        next
+ * @optionparent plotOptions.dumbbell
+ */
+seriesType<Highcharts.DumbbellSeries>('dumbbell', 'arearange', {
+    /** @ignore-option */
+    trackByArea: false,
+    /** @ignore-option */
+    fillColor: 'none',
+    /** @ignore-option */
+    lineWidth: 0,
+    pointRange: 1,
+    /**
+     * Pixel width of the line that connects dumbbell point's values.
+     *
+     * @since     next
+     * @product   highcharts highstock
+     */
+    connectorWidth: 1,
+    /** @ignore-option */
+    stickyTracking: false,
+    groupPadding: 0.2,
+    pointPadding: 0.1,
+    /**
+     * Color of the start markers in dumbbell graph.
+     *
+     * @type      {Highcharts.ColorString|Highcharts.GradientColorObject|Highcharts.PatternObject}
+     * @since     next
+     * @product   highcharts highstock
+     */
+    startColor: '${palette.neutralColor80}',
+    /**
+     * Color of the line that connects dumbbell point's values.
+     * By default it is a series' color.
+     *
+     * @type      {string}
+     * @product   highcharts highstock
+     * @since     next
+     * @apioption plotOptions.dumbbell.connectorColor
+     */
+    states: {
+        hover: {
+            /** @ignore-option */
+            lineWidthPlus: 0,
+            /**
+             * The additional connector line width for a hovered point.
+             *
+             * @since next
+             * @product   highcharts highstock
+             */
+            connectorWidthPlus: 1,
+            /** @ignore-option */
+            halo: false
+        }
+    }
+}, {
+    trackerGroups: ['group', 'markerGroup', 'dataLabelsGroup'],
+    drawTracker: H.TrackerMixin.drawTrackerPoint,
+    drawGraph: H.noop,
+    /**
+     * Correct line position by Math.floor instead of round.
+     * As a result the line is aligned in the same way as marker
+     *
+     * @private
+     *
+     * @function Highcharts.seriesTypes.dumbbell#crispConnector
+     *
+     * @param {Highcharts.SVGRenderer} this
+     *        Highcharts Renderer.
+     * @param {Highcharts.SVGPathArray} points
+     *        The original points on the format `['M', 0, 0, 'L', 100, 0]`.
+     *
+     * @return {Highcharts.SVGPathArray}
+     *         The original points array, but modified to render crisply.
+     *
+     *
+     */
+    crispConnector: function (
+        this: Highcharts.SVGRenderer,
+        points: Highcharts.SVGPathArray
+    ): Highcharts.SVGPathArray {
+        if (points[1] === points[4]) {
+            // Substract due to #1129. Now bottom and left axis gridlines behave
+            // the same.
+            points[1] = points[4] = Math.floor(points[1] as any) - 0.5;
```

Comment:
Substracting 0.5 works well when the line width is an odd number, but not with even numbers: https://jsfiddle.net/highcharts/t5wdcqyg/. See crisping logic elsewhere in the codebase, where we use the modulo operator.